src/huggingface_hub/hf_api.py

```diff
@@ -4517,27 +4591,39 @@ def upload_folder(
                 merge_pr=not create_pr,
                 verbose=multi_commits_verbose,
             )
-        else:
-            commit_info = self.create_commit(
-                repo_type=repo_type,
-                repo_id=repo_id,
-                operations=commit_operations,
-                commit_message=commit_message,
-                commit_description=commit_description,
-                token=token,
-                revision=revision,
-                create_pr=create_pr,
-                parent_commit=parent_commit,
-            )
-            pr_url = commit_info.pr_url
+            # Defining a CommitInfo object is not really relevant in this case
+            # Let's return early with pr_url only (as string).
+            return pr_url
```

Comment:
Returning a list of commits in the multi-commit case would make more sense, no?

Reply:
In general `multi_commits` is still flagged as `"experimental"` and supposedly not much used so I prefer not to touch in this PR to keep it simple. In any case, we can update the type in the future without prior notice. Returning a list of commits is a good idea though. We would need an extra call to list the PR commits but that's straightforward.